src/docfx/lib/path/PathUtility.cs

```diff
@@ -85,6 +88,27 @@ public static string NormalizeFolder(string path)
         return str;
     }
 
+    public static string NormalizeFileWithValidation(string path)
```

Comment:
Keep the function single purpose, the caller can chain these two functions
```suggestion
    public static string CheckInvalidPathString(string path)
```